packages/twenty-server/src/core/workspace/workspace.resolver.ts

```diff
@@ -67,6 +69,13 @@ export class WorkspaceResolver {
 
   @Mutation(() => Workspace)
   async deleteCurrentWorkspace(@AuthWorkspace() { id }: Workspace) {
+    const demoWorkspaceIds = this.environmentService.getDemoWorkspaceIds();
+
+    // Check if the id is in the list of demo workspaceIds
+    if (demoWorkspaceIds.includes(id)) {
+      throw new Error('Cannot delete a demo workspace.');
```

Comment:
@khakimov could you throw a `ForbiddenException` instead? This will (ongoing work) be parsed by our exception handler and nicely sent back through graphql response.

Reply:
added checks in deleteUser as well to prevent removing current user from the demo workspace.